### PR TITLE
Re-attempt for taskbar compensation

### PR DESCRIPTION
**Pull request by WPFerg**
#451

### CODE REVIEW

File: src/window-service.js [hunk 1]
```diff
@@ -196,20 +196,53 @@
...
+                        var midY = monitor.monitorRect.top + monitor.monitorRect.bottom / 2;
+                        var distance = Math.pow(midX - x, 2) + Math.pow(midY - y, 2);
+                        if (distance < closestDistance) {
+                            closestDistance = distance;
+                            closestMonitor = monitor;
+                        }
+
+                    }
+                }
+                callback(closestMonitor);
```
Comment (owennw) [inline src/window-service.js]: Maybe `monitor.monitorRect` should be a variable? And is there anything in the geometryService (or that should be added to it) that could help out here?

File: src/window-service.js [hunk 2]
```diff
@@ -196,20 +196,53 @@
...
                     }
 
-                    if (bounds.top < info.virtualScreen.top) {
-                        bounds.top = info.virtualScreen.top;
-                    } else if (bounds.top + bounds.height > info.virtualScreen.bottom) {
-                        bounds.top = info.virtualScreen.bottom - bounds.height;
+                    if (bounds.top < monitor.availableRect.top) {
+                        bounds.top = monitor.availableRect.top;
+                    } else if (bounds.top + bounds.height > monitor.availableRect.bottom) {
+                        bounds.top = monitor.availableRect.bottom - bounds.height;
```
Comment (owennw) [inline src/window-service.js]: Similarly maybe `monitor.availableRect` should be a variable.

File: src/window-service.js (hunk 1 above)
Comment (WPFerg) [inline src/window-service.js]: Possibly, the `intersects` function there is pretty similar, but `Rectangle`'s constructor requires `left, top, width, height`, opposed to `left, top, bottom, right`, so it mightn't be worth the overhead converting for quite a simple operation.

Comment (owennw) [inline src/window-service.js]: Fair enough. It does feel like a function that belongs in the geometry area, but it's not a big deal for the size and scope of the app.

**Pull request merged**